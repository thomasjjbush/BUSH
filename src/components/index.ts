export { AspectRatio } from './aspect-ratio/aspect-ratio';
export { Button } from './button/button';
export { Carousel } from './carousel/carousel';
export { Client } from './client/client';
export { Error } from './error/error';
export { HiddenText } from './hidden-text/hidden-text';
export { Loading } from './loading/loading';
export { Icon } from './icon/icon';
export { Image } from './image/image';
export { Nav } from './nav/nav';
export { Scroller } from './scroller/scroller';
export { Tags } from './tags/tags';
export { Thumbnail } from './thumbnail/thumbnail';
