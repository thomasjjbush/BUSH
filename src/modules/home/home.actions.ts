import { GraphQLClient } from 'graphql-request';
import { batch } from 'react-redux';
import { Dispatch } from 'redux';
import { ClientsActions, EmploymentActions, ProjectsActions, TagActions } from '../../types';
import { homePageQuery, projectsQuery } from './home.queries';

export const loadHomePage = (graphQL: GraphQLClient) => async (dispatch: Dispatch): Promise<void> => {
    try {
        const { clients, employment, projects, tags } = await graphQL.request(homePageQuery);
        batch(() => {
            dispatch({ type: ClientsActions.LOADED, payload: clients.items });
            dispatch({ type: EmploymentActions.LOADED, payload: employment.items });
            dispatch({ type: ProjectsActions.LOADED, payload: projects });
            dispatch({ type: TagActions.LOADED, payload: tags.items });
        });
    } catch (e) {
        console.log(e);
    }
};

export const loadProjects = (
    graphQL: GraphQLClient,
    args: { client?: string; order?: string; search?: string; skip?: number },
) => async (dispatch: Dispatch): Promise<void> => {
    try {
        if (args.skip) dispatch({ type: ProjectsActions.LOADING_MORE });

        const { projects } = await graphQL.request(projectsQuery, args);

        dispatch({
            type: args.skip ? ProjectsActions.LOADED_MORE : ProjectsActions.LOADED,
            payload: projects,
        });
    } catch (e) {
        console.error(e);
    }
};
