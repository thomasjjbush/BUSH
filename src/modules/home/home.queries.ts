import { gql } from 'graphql-request';
import { clientFragment, projectFragment, tagFragment } from '../../utils';

export const homePageQuery = gql`
    query {
        clients: clientsCollection(order: [name_ASC]) {
            items {
                ...ClientFragment
            }
        }
        employment: employmentCollection(order: [startDate_DESC]) {
            items {
                companyName
                endDate
                location {
                    lon
                    lat
                }
                responsibilities
                startDate
                title
                url
            }
        }
        projects: projectCollection(limit: 4, order: [year_DESC]) {
            items {
                ...ProjectFragment
                thumbnail {
                    url
                }
            }
            total
        }
        tags: tagCollection(order: [name_ASC]) {
            items {
                ...TagFragemnt
            }
        }
    }
    ${clientFragment}
    ${projectFragment}
    ${tagFragment}
`;

export const projectsQuery = gql`
    query($client: String = "", $order: ProjectOrder = year_DESC, $search: String = "", $skip: Int = 0) {
        projects: projectCollection(
            limit: 4
            order: [$order]
            skip: $skip
            where: { AND: [{ client: { name_contains: $client } }, { name_contains: $search }] }
        ) {
            items {
                ...ProjectFragment
                thumbnail {
                    url
                }
            }
            total
        }
    }
    ${projectFragment}
`;
