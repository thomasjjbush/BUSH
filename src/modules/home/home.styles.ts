import styled from 'styled-components';
import { StyledProps } from '../../types';

export const CV = styled.section<StyledProps>`
    align-self: center;
    flex: 1;
    display: flex;
    flex-direction: column;
    flex: 1;
    overflow: hidden;
    background-color: ${(props): string => props.theme.colors.offset};
    padding: 30px;
`;

export const CVContainer = styled.section`
    display: flex;
    flex-direction: column;
    max-width: 500px;

    @media only screen and (max-width: ${(props): number => props.theme.breakpoints.tabletPortrait}px) {
        flex: 1 0 0;
        margin-bottom: 30px;
        margin-right: 0 !important;
        max-height: 40%;
        max-width: none;
    }

    @media only screen and (max-width: ${(props): number => props.theme.breakpoints.mobile}px) {
        display: none;
    }
`;

export const CVInfo = styled.div`
    margin-bottom: 30px;
`;

export const CVTitle = styled.p`
    margin: 0 0 30px 0;
`;

export const Date = styled.p`
    margin: 0;
`;

export const Experience = styled.div`
    display: grid;
    grid-template-columns: 25% 30px auto;
`;

export const ExperienceInfo = styled.div`
    flex: 1;
    padding-left: 30px;
`;

export const Filters = styled.div`
    display: flex;
    flex-shrink: 0;
    margin-bottom: 15px;
`;

export const Grid = styled.div`
    display: grid;
    grid-template-columns: calc(50% - 5px) calc(50% - 5px);
    grid-gap: 10px;
    margin: 1px;

    @media only screen and (max-width: ${(props): number => props.theme.breakpoints.mobile}px) {
        grid-template-columns: auto;
        grid-gap: 30px;
    }
`;

export const Input = styled.input<StyledProps>`
    background-color: ${(props): string => props.theme.colors.highlight};
    border: solid 2px ${(props): string => props.theme.colors.border};
    border-radius: 5px;
    color: ${(props): string => props.theme.colors.text};
    flex: 1;
    font-size: 16px;
    height: 40px;
    padding: 0 10px;
`;

export const Projects = styled.section`
    display: flex;
    flex-direction: column;
    overflow: visible;
`;

export const Select = styled.select<StyledProps>`
    appearance: none;
    background-color: ${(props): string => props.theme.colors.highlight};
    border: solid 2px ${(props): string => props.theme.colors.border};
    border-radius: 5px;
    color: ${(props): string => props.theme.colors.text};
    flex-shrink: 0;
    font-size: 16px;
    margin-left: 15px;
    padding: 0 5px;
    width: 60px;
`;

export const Section = styled.section`
    flex: 1;
`;

export const Timeline = styled.div<StyledProps>`
    :before,
    :after {
        background-color: ${(props): string => props.theme.colors.brand};
        content: '';
        display: block;
        margin: auto;
    }
    :before {
        border-radius: 50%;
        height: 20px;
        width: 20px;
    }

    :after {
        height: 100%;
        width: 2px;
    }
`;
