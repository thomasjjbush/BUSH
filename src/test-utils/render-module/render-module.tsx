import React, { FunctionComponent } from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { mount, ReactWrapper } from 'enzyme';
import { mockStore } from '../../test-utils/mock-store/mock-store';
import { ThemeProvider } from 'styled-components';
import { mockTheme } from '../../test-utils/mock-theme/mock-theme';
import { initTheme } from '../../modules/theme/theme';
import { Store } from '../../types';

const client = {
    logo: {
        url: 'client-logo-url',
    },
    name: 'client-name',
    primaryColor: 'client-primaryColor',
    slug: 'client-slug',
    url: 'client-url',
};

const project = {
    description: 'project-description',
    hero: {
        url: 'project-hero-url',
    },
    client,
    gallery: {
        isLandScape: false,
        total: 1,
        items: [
            {
                url: 'project-gallery-0-url',
            },
        ],
    },
    name: 'project-name',
    primaryTag: { name: 'primary-tag-name', slug: 'primary-tag-slug' },
    responsibilities: {
        items: [
            {
                description: 'responsibility-description',
                icon: 'responsibility-icon',
                name: 'responsibility-name',
            },
        ],
    },
    slug: 'project-slug',
    tags: { items: [{ name: 'tag-name', slug: 'tag-slug' }] },
    thumbnail: {
        name: 'x',
        url: 'project-thumbnail-url',
    },
    year: 1970,
};

const defaultState: Store = {
    clients: [client],
    employment: [
        {
            companyName: 'company',
            endDate: new Date(null).toDateString(),
            location: { lat: 'lat', lon: 'lon' },
            responsibilities: 'responsibilities',
            startDate: new Date(null).toDateString(),
            title: 'title',
            url: 'url',
        },
    ],
    project: {
        item: project,
        sameClient: [project],
        sameTag: [project],
    },
    projects: {
        items: [project],
        total: 10,
    },
    tags: [{ name: 'tag-name', slug: 'tag-slug' }],
};

export const renderModule = (Component: FunctionComponent, overrides = {}): ReactWrapper =>
    mount(
        <Provider store={mockStore({ ...defaultState, ...overrides })}>
            <ThemeProvider theme={mockTheme(initTheme(false, true) as any)}>
                <Router history={createMemoryHistory()}>
                    <Component />
                </Router>
            </ThemeProvider>
        </Provider>,
    );
