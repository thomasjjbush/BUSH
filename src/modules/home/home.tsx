import React, { ChangeEvent, FunctionComponent, ReactElement, useEffect, useRef, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { loadHomePage, loadProjects } from './home.actions';
import { Store } from '../../types';
import { labels, orders } from '../../config';
import * as Styled from './home.styles';
import { Link } from 'react-router-dom';
import { StyledMain } from '../theme/global';
import { Client, Scroller, Tags, Thumbnail } from '../../components';
import { getMonthAndYear, useGraphQL } from '../../utils/';

const renderOption = (name: string, val?: string | number): ReactElement => (
    <option key={val ?? name} value={val ?? name}>
        {name}
    </option>
);

const Home: FunctionComponent = (): ReactElement => {
    const graphQL = useGraphQL();
    const dispatch = useDispatch();
    const { clients, employment, projects, tags } = useSelector<Store, Partial<Store>>(state => ({
        clients: state.clients,
        employment: state.employment,
        projects: state.projects,
        tags: state.tags,
    }));

    const cv = useRef(null);
    const [client, setClient] = useState('');
    const [order, setOrder] = useState(orders[0].value);
    const [search, setSearch] = useState('');

    useEffect(() => {
        dispatch(loadHomePage(graphQL));
    }, []);

    const onChange = (cb: Function, key: string) => (e: ChangeEvent<HTMLInputElement | HTMLSelectElement>): void => {
        cb(e.target.value);
        dispatch(loadProjects(graphQL, { client, order, search, [key]: e.target.value }));
    };

    const onClick = (): void => {
        dispatch(loadProjects(graphQL, { client, order, search, skip: projects.items.length }));
    };

    return (
        <StyledMain max={2000} padding={30}>
            <Styled.CVContainer>
                <Styled.CV ref={cv}>
                    <Styled.CVInfo>
                        <h1>{labels.name}</h1>
                        <h2>{labels.jobTitle}</h2>
                        <Tags tags={tags} />
                    </Styled.CVInfo>

                    <Scroller>
                        <div>
                            <Styled.CVTitle>{labels.aboutMe}</Styled.CVTitle>
                            <h2>{labels.experience}</h2>
                            {employment.map(({ companyName, endDate, responsibilities, startDate, title }, i) => (
                                <Styled.Experience key={i}>
                                    <Styled.Date>
                                        {getMonthAndYear(startDate)} - {getMonthAndYear(endDate) || labels.present}
                                    </Styled.Date>
                                    <Styled.Timeline />
                                    <Styled.ExperienceInfo>
                                        <h2>{title}</h2>
                                        <h3>{companyName}</h3>
                                        <p>{responsibilities}</p>
                                    </Styled.ExperienceInfo>
                                </Styled.Experience>
                            ))}
                        </div>
                    </Scroller>
                </Styled.CV>
            </Styled.CVContainer>
            <Styled.Projects>
                <h1>
                    Projects ({projects.items.length}/{projects.total || 0})
                </h1>
                <Styled.Filters>
                    <Styled.Input
                        aria-label={labels.search}
                        data-test-id="search-input"
                        onChange={onChange(setSearch, 'search')}
                        placeholder={labels.search}
                        value={search}
                        type="text"
                    />
                    <Styled.Select
                        aria-label={labels.selectOrder}
                        data-test-id="select-order"
                        onChange={onChange(setOrder, 'order')}
                        value={order}
                    >
                        {orders.map(({ name, value }) => renderOption(name, value))}
                    </Styled.Select>
                    <Styled.Select
                        aria-label={labels.filterByClient}
                        data-test-id="select-client"
                        onChange={onChange(setClient, 'client')}
                        value={client}
                    >
                        {renderOption('All', '')}
                        {clients.map(({ name }) => renderOption(name))}
                    </Styled.Select>
                </Styled.Filters>
                <Scroller
                    loadMore={onClick}
                    numberOfItems={projects.items.length}
                    loading={projects.loading}
                    totalItems={projects.total}
                    testID="project-scroller"
                >
                    <Styled.Grid>
                        {projects.items.map(({ client, name, slug, tags, thumbnail }) => (
                            <Link data-test-id="project" key={slug} to={`/projects/${slug}`}>
                                <Thumbnail
                                    client={<Client alt={client.name} logo={client.logo.url} size={200} width={100} />}
                                    color={client.primaryColor}
                                    name={name}
                                    url={thumbnail.url}
                                />
                                <Tags tags={tags.items} />
                            </Link>
                        ))}
                    </Styled.Grid>
                </Scroller>
            </Styled.Projects>
        </StyledMain>
    );
};

Home.displayName = 'Home';

export default Home;
