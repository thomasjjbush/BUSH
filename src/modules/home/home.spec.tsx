import { ReactNode } from 'react';
import { ReactWrapper } from 'enzyme';
import { act } from 'react-dom/test-utils';
import Home from './home';
import { loadHomePage, loadProjects } from './home.actions';
import { Input, Select } from './home.styles';
import { Scroller } from '../../components';
import { renderModule } from '../../test-utils/render-module/render-module';

jest.mock('../../utils', () => ({
    ...jest.requireActual('../../utils'),
    useGraphQL: jest.fn().mockReturnValue('graphQLContext'),
}));

jest.mock('./home.actions', () => ({
    loadHomePage: jest.fn().mockReturnValue({ type: 'MOCK_LOAD_HOME_PAGE' }),
    loadProjects: jest.fn().mockReturnValue({ type: 'MOCK_LOAD_PROJECTS' }),
}));

jest.mock('../../components', () => ({
    Client: (): string => '',
    Scroller: ({ children }: { children: ReactNode[] }): ReactNode => children as ReactNode,
    Tags: (): string => '',
    Thumbnail: (): string => '',
}));

describe('Home', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it('should match snapshot', () => {
        expect(renderModule(Home).find(Home)).toMatchSnapshot();
    });

    it('should invoke loadHomePage on mount', () => {
        renderModule(Home);
        expect(loadHomePage).toHaveBeenCalledTimes(1);
    });

    it.each`
        action             | component | expectedArgs                                          | initialState   | selector
        ${'search input'}  | ${Input}  | ${{ client: '', order: 'year_DESC', search: 'xoxo' }} | ${''}          | ${'first'}
        ${'order select'}  | ${Select} | ${{ client: '', order: 'xoxo', search: '' }}          | ${'year_DESC'} | ${'first'}
        ${'client select'} | ${Select} | ${{ client: 'xoxo', order: 'year_DESC', search: '' }} | ${''}          | ${'last'}
    `(
        'should update state and dispatch action on $action chane',
        ({
            component,
            expectedArgs,
            initialState,
            selector,
        }: {
            component: ReactWrapper;
            expectedArgs: Record<string, string>;
            initialState: string;
            selector: 'first' | 'last';
        }) => {
            const wrapper = renderModule(Home);
            const elem = wrapper.find(component)[selector]();
            expect(elem.prop('value')).toBe(initialState);

            act(() => {
                elem.prop('onChange')({ target: { value: 'xoxo' } });
            });
            wrapper.update();

            expect(
                wrapper
                    .find(component)
                    [selector]()
                    .prop('value'),
            ).toBe('xoxo');
            expect(loadProjects).toHaveBeenCalledTimes(1);
            expect(loadProjects).toHaveBeenCalledWith('graphQLContext', expectedArgs);
        },
    );

    it('should load more projects on click', () => {
        const wrapper = renderModule(Home);
        wrapper
            .find(Scroller)
            .last()
            .prop('loadMore')();

        expect(loadProjects).toHaveBeenCalledTimes(1);
        expect(loadProjects).toHaveBeenCalledWith('graphQLContext', {
            client: '',
            order: 'year_DESC',
            search: '',
            skip: 1,
        });
    });
});
