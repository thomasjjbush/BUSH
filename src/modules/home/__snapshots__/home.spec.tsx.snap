// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<Home>
  <styled.main
    max={2000}
    padding={30}
    styles="
    display: flex;
    flex-direction: auto;
    flex: 1;
    overflow: hidden;
    padding: 30px;
    position: relative;
    margin: auto;
    max-width: 2000px;
    width: 100%;

    @media only screen and (max-width: theme.breakpoints.tabletPortraitpx) {
        flex-direction: column;
    }
"
  >
    <main
      className="sc-dQppl cKGjcR"
      max={2000}
    >
      <styled.section
        styles="
    display: flex;
    flex-direction: column;
    max-width: 500px;

    @media only screen and (max-width: theme.breakpoints.tabletPortraitpx) {
        flex: 1 0 0;
        margin-bottom: 30px;
        margin-right: 0 !important;
        max-height: 40%;
        max-width: none;
    }

    @media only screen and (max-width: theme.breakpoints.mobilepx) {
        display: none;
    }
"
      >
        <section
          className="sc-gsTCUz dVXaAZ"
        >
          <styled.section
            styles="
    align-self: center;
    flex: 1;
    display: flex;
    flex-direction: column;
    flex: 1;
    overflow: hidden;
    background-color: theme.colors.offset;
    padding: 30px;
"
          >
            <section
              className="sc-bdfBwQ ioSnma"
            >
              <styled.div
                styles="
    margin-bottom: 30px;
"
              >
                <div
                  className="sc-dlfnbm kCPdvt"
                >
                  <h1>
                    laebls.name
                  </h1>
                  <h2>
                    laebls.jobTitle
                  </h2>
                  <Tags
                    tags={
                      Array [
                        Object {
                          "name": "tag-name",
                          "slug": "tag-slug",
                        },
                      ]
                    }
                  />
                </div>
              </styled.div>
              <Scroller>
                <div>
                  <styled.p
                    styles="
    margin: 0 0 30px 0;
"
                  >
                    <p
                      className="sc-hKgILt hynpnB"
                    >
                      laebls.aboutMe
                    </p>
                  </styled.p>
                  <h2>
                    laebls.experience
                  </h2>
                  <styled.div
                    key="0"
                    styles="
    display: grid;
    grid-template-columns: 25% 30px auto;
"
                  >
                    <div
                      className="sc-jSgupP iXrUPc"
                    >
                      <styled.p
                        styles="
    margin: 0;
"
                      >
                        <p
                          className="sc-eCssSg bRCGol"
                        >
                          Jan 1970
                           - 
                          Jan 1970
                        </p>
                      </styled.p>
                      <styled.div
                        styles="
    :before,
    :after {
        background-color: theme.colors.brand;
        content: '';
        display: block;
        margin: auto;
    }
    :before {
        border-radius: 50%;
        height: 20px;
        width: 20px;
    }

    :after {
        height: 100%;
        width: 2px;
    }
"
                      >
                        <div
                          className="sc-crrsfI hoTnwh"
                        />
                      </styled.div>
                      <styled.div
                        styles="
    flex: 1;
    padding-left: 30px;
"
                      >
                        <div
                          className="sc-gKsewC cEJyVK"
                        >
                          <h2>
                            title
                          </h2>
                          <h3>
                            company
                          </h3>
                          <p>
                            responsibilities
                          </p>
                        </div>
                      </styled.div>
                    </div>
                  </styled.div>
                </div>
              </Scroller>
            </section>
          </styled.section>
        </section>
      </styled.section>
      <styled.section
        styles="
    display: flex;
    flex-direction: column;
    overflow: visible;
"
      >
        <section
          className="sc-jrAGrp jjoPaf"
        >
          <h1>
            Projects (
            1
            /
            10
            )
          </h1>
          <styled.div
            styles="
    display: flex;
    flex-shrink: 0;
    margin-bottom: 15px;
"
          >
            <div
              className="sc-iBPRYJ ftqSnZ"
            >
              <styled.input
                aria-label="laebls.search"
                data-test-id="search-input"
                onChange={[Function]}
                placeholder="laebls.search"
                styles="
    background-color: theme.colors.highlight;
    border: solid 2px theme.colors.border;
    border-radius: 5px;
    color: theme.colors.text;
    flex: 1;
    font-size: 16px;
    height: 40px;
    padding: 0 10px;
"
                type="text"
                value=""
              >
                <input
                  aria-label="laebls.search"
                  className="sc-pFZIQ fzTwpq"
                  data-test-id="search-input"
                  onChange={[Function]}
                  placeholder="laebls.search"
                  type="text"
                  value=""
                />
              </styled.input>
              <styled.select
                aria-label="laebls.selectOrder"
                data-test-id="select-order"
                onChange={[Function]}
                styles="
    appearance: none;
    background-color: theme.colors.highlight;
    border: solid 2px theme.colors.border;
    border-radius: 5px;
    color: theme.colors.text;
    flex-shrink: 0;
    font-size: 16px;
    margin-left: 15px;
    padding: 0 5px;
    width: 60px;
"
                value="year_DESC"
              >
                <select
                  aria-label="laebls.selectOrder"
                  className="sc-kEjbxe efkPQT"
                  data-test-id="select-order"
                  onChange={[Function]}
                  value="year_DESC"
                >
                  <option
                    key="year_DESC"
                    value="year_DESC"
                  >
                    Latest
                  </option>
                  <option
                    key="year_ASC"
                    value="year_ASC"
                  >
                    Oldest
                  </option>
                </select>
              </styled.select>
              <styled.select
                aria-label="laebls.filterByClient"
                data-test-id="select-client"
                onChange={[Function]}
                styles="
    appearance: none;
    background-color: theme.colors.highlight;
    border: solid 2px theme.colors.border;
    border-radius: 5px;
    color: theme.colors.text;
    flex-shrink: 0;
    font-size: 16px;
    margin-left: 15px;
    padding: 0 5px;
    width: 60px;
"
                value=""
              >
                <select
                  aria-label="laebls.filterByClient"
                  className="sc-kEjbxe efkPQT"
                  data-test-id="select-client"
                  onChange={[Function]}
                  value=""
                >
                  <option
                    key=""
                    value=""
                  >
                    All
                  </option>
                  <option
                    key="client-name"
                    value="client-name"
                  >
                    client-name
                  </option>
                </select>
              </styled.select>
            </div>
          </styled.div>
          <Scroller
            loadMore={[Function]}
            numberOfItems={1}
            testID="project-scroller"
            totalItems={10}
          >
            <styled.div
              styles="
    display: grid;
    grid-template-columns: calc(50% - 5px) calc(50% - 5px);
    grid-gap: 10px;
    margin: 1px;

    @media only screen and (max-width: theme.breakpoints.mobilepx) {
        grid-template-columns: auto;
        grid-gap: 30px;
    }
"
            >
              <div
                className="sc-fubCfw MaiIs"
              >
                <Link
                  data-test-id="project"
                  key="project-slug"
                  to="/projects/project-slug"
                >
                  <LinkAnchor
                    data-test-id="project"
                    href="/projects/project-slug"
                    navigate={[Function]}
                  >
                    <a
                      data-test-id="project"
                      href="/projects/project-slug"
                      onClick={[Function]}
                    >
                      <Thumbnail
                        client={
                          <Client
                            alt="client-name"
                            logo="client-logo-url"
                            size={200}
                            width={100}
                          />
                        }
                        color="client-primaryColor"
                        name="project-name"
                        url="project-thumbnail-url"
                      />
                      <Tags
                        tags={
                          Array [
                            Object {
                              "name": "tag-name",
                              "slug": "tag-slug",
                            },
                          ]
                        }
                      />
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </styled.div>
          </Scroller>
        </section>
      </styled.section>
    </main>
  </styled.main>
</Home>
`;
